firebase-firestore/src/main/java/com/google/firebase/firestore/model/Values.java

```diff
@@ -432,4 +457,64 @@ public static Value refValue(DatabaseId databaseId, DocumentKey key) {
             .build();
     return value;
   }
+
+  public static Value getFirstValue(Value.ValueTypeCase valueTypeCase) {
+    switch (valueTypeCase) {
+      case NULL_VALUE:
+        return Values.NULL_VALUE;
+      case BOOLEAN_VALUE:
+        return Value.newBuilder().setBooleanValue(false).build();
+      case INTEGER_VALUE:
+      case DOUBLE_VALUE:
+        return Value.newBuilder().setDoubleValue(Double.NaN).build();
+      case TIMESTAMP_VALUE:
+        return Value.newBuilder()
+            .setTimestampValue(Timestamp.newBuilder().setSeconds(Long.MIN_VALUE))
+            .build();
+      case STRING_VALUE:
+        return Value.newBuilder().setStringValue("").build();
+      case BYTES_VALUE:
+        return Value.newBuilder().setBytesValue(ByteString.EMPTY).build();
+      case REFERENCE_VALUE:
+        return refValue(DatabaseId.EMPTY, DocumentKey.empty());
+      case GEO_POINT_VALUE:
+        return Value.newBuilder()
+            .setGeoPointValue(LatLng.newBuilder().setLatitude(-90.0).setLongitude(-180.0))
+            .build();
+      case ARRAY_VALUE:
+        return Value.newBuilder().setArrayValue(ArrayValue.getDefaultInstance()).build();
+      case MAP_VALUE:
+        return Value.newBuilder().setMapValue(MapValue.getDefaultInstance()).build();
+      default:
+        throw new IllegalArgumentException("Unknown value type: " + valueTypeCase);
+    }
+  }
+
+  public static @Nullable Value getNextValue(Value.ValueTypeCase valueTypeCase) {
```

Comment:
s/getNextValue/getHighestValue or getHighestPossibleValue?

NextValue is a bit confusing here. Alternatively, just add some code comments please.

Reply:
Looks like I forgot to push the commit with the comments. I agree that this is not a great name - however this method returns an exclusive value, i.e. one higher than the max. Do you have a name suggestion that conveys that? I need your input here :)

Reply:
Some ideas: getHighestValueExclusive, getUpperBoundExclusive, getExclusiveUpperBound

I think pairing "exclusive" with a "bound" makes more sense. Or you can leave the current name and just have comments.

Reply:
Changed to getLowerBound() and getUpperBound(). The comments still state whether it is exclusive or not.